src/consensus/pbft/pbft.h

```diff
@@ -372,11 +478,110 @@ namespace pbft
       switch (serialized::peek<PbftMsgType>(data, size))
       {
         case pbft_message:
+        {
           serialized::skip(data, size, sizeof(PbftHeader));
           message_receiver_base->receive_message(data, size);
           break;
-        default:
-        {}
+        }
+        case pbft_append_entries:
+        {
+          AppendEntries r;
+
+          auto append_entries_index = store->current_version();
```

Comment:
We should guarantee that no transaction can be committed on that node when it is catching up (i.e. frontend closed, etc.) otherwise this will get out of sync.

Reply:
If you look at `handle(Pre_prepare)`, `handle(Prepare)` and `handle(Commit)` I reject any messages if we are playing back, so nothing else can be executed.